### PR TITLE
Add option to print file being deleted (-V or --verbose)

### LINKED ISSUE #140: Verbose Flag (for CLI)
I was surprised to find that the verbose flag (`-v` or `--verbose`) was not supported, whereas it is within `rm`.

While this functionality could be considered as a separate package surrounding `rimraf`, the files being deleted are not exposed, meaning it would have to include its own globbing logic, handing single files to `rimraf` to delete and thereby losing much of the benefit of using it in the first place.

I appreciate the desire to keep any sort of UI out of `rimraf`, but given that `rm` supports `-v` (lines of `removed foo/bar/baz.txt`) and that `rimraf` advertises itself as a substitute for `rm -rf`, I figured it was worth a conversation.

### PR DESCRIPTION
**Pull request by brunolm**
Closes #140 

```
rimraf -V <args>
rimraf --verbose <args>
```

### CODE REVIEW

**Comment by rashidul0405** [comment]
@isaacs @brunolm Is there any chance of this to get merge?

**Comment by isaacs** [comment]
It would be better if the lib wasn't printing.  Only the bin should print, the lib should return or stream the files being deleted, and the bin should print them if `--verbose` is set.

File: bin.js [hunk 1]
```diff
@@ -32,16 +36,23 @@ if (help || args.length === 0) {
   log('  -h, --help     Display this usage info')
   log('  -G, --no-glob  Do not expand glob patterns in arguments')
   log('  -g, --glob     Expand glob patterns in arguments (default)')
+  log('  -V, --verbose  Prints paths being deleted')
```
Comment (ajitzero) [inline bin.js:39]: The `rm` command supports `-v` (lowercase) instead of `-V`. We could support both.

File: rimraf.js [hunk 2]
```diff
@@ -107,6 +108,9 @@ function rimraf (p, options, cb) {
           if (er.code === "ENOENT") er = null
         }
 
+        if (options.verbose)
+          console.log('deleting', p)
```
Comment (ajitzero) [inline rimraf.js:112]: fwiw, we should use `console.info`

**Review by ajitzero — CHANGES_REQUESTED** [review]

File: rimraf.js (hunk 2 above)
Comment (isaacs) [reply rimraf.js:112]: Why?  It's just writing a string to stdout, in this case they're 100% equivalent, I almost always use `console.log()` for this purpose.

Comment (ajitzero) [reply rimraf.js:112]: A lot of terminal profiles can be (and are) setup to visually highlight different forms of logs. afaik there is also a screen reader for terminal output which can differentiate between log & info.

Comment (ajitzero) [reply rimraf.js:112]: You may dismiss this suggestion if it isn't aligned with other code here. I did prefix this with fwiw - not a deal breaker for most.

Comment (isaacs) [reply rimraf.js:112]: They are literally the same function: https://github.com/nodejs/node/blob/main/lib/internal/console/constructor.js#L682

The only way you could highlight different forms of logs is by patching the global `console` object.

They're displayed differently in the browser, but in Node, they're 100% identical, just same function with two different names.

**Comment by isaacs** [comment]
Implemented somewhat differently on https://github.com/isaacs/rimraf/commit/3b57687ac28c7feee2caae455fb52d1b55e39f92, using the `filter` option added in 4.3.

**Pull request closed without merging**